frontend/src/views/File.vue

```diff
@@ -502,11 +507,24 @@ export default class FileView extends Vue {
     await this.fetchFileMetadata(payload);
     if (this.response == null || this.error) return;
     await Promise.all([
+      this.fetchInstrument(this.response),
       this.fetchVisualizations(payload),
       this.fetchVersions(this.response),
       this.fetchSourceFiles(this.response),
     ]);
     this.isBusy = false;
   }
+
+  async fetchInstrument(file: RegularFile | File) {
+    if (!("instrumentPid" in file)) return;
```

Comment:
Pitää tarkistaa myös, ettei `instrumentPid` ole `null`. Kummatkin tarkistukset voi yhdistää:

```suggestion
    if (!file.instrumentPid) return;
```

Reply:
Totta, `null` pitää tarkastaa. Ehdotuksesi kyllä toimii, mutta typescript kääntäjä valittaa siitä. Ehkä on paras vaan kirjoittaa
`if (!("instrumentPid" in file) || file.instrumentPid == null) return;` vaikka se onkin vähän verboosia. Tyyppi pitäisi muuten olla `Regularfile | Modelfile`, pitää korjata sekin..